// ng-orderby-pipe/fesm2015/ng-orderby-pipe.js
import { __decorate } from 'tslib';
import { Pipe, NgModule } from '@angular/core';

let OrderByPipe = class OrderByPipe {
    transform(values, objectKeyWithPath = '', order = 'asc') {
        const predicatePathValues = objectKeyWithPath.split('.');
        const desc = order === 'desc';
        if (!objectKeyWithPath) {
            return desc ? values.sort().reverse() : values.sort();
        }
        return values.sort((a, b) => {
            let result = 0;
            let leftSideValue = a;
            let rightSideValue = b;
            predicatePathValues.forEach((predicate) => {
                leftSideValue = leftSideValue[predicate];
                rightSideValue = rightSideValue[predicate];
            });
            if (leftSideValue < rightSideValue) {
                result = desc ? 1 : -1;
            }
            else if (rightSideValue < leftSideValue) {
                result = desc ? -1 : 1;
            }
            return result;
        });
    }
};
OrderByPipe = __decorate([
    Pipe({ name: 'orderBy' })
], OrderByPipe);

let OrderByModule = class OrderByModule {
};
OrderByModule = __decorate([
    NgModule({
        declarations: [OrderByPipe],
        exports: [OrderByPipe],
        providers: [OrderByPipe]
    })
], OrderByModule);

/**
 * Generated bundle index. Do not edit.
 */

export { OrderByModule, OrderByPipe };
