// ng-orderby-pipe/fesm5/ng-orderby-pipe.js
import { __decorate } from 'tslib';
import { Pipe, NgModule } from '@angular/core';

var OrderByPipe = /** @class */ (function () {
    function OrderByPipe() {
    }
    OrderByPipe.prototype.transform = function (values, objectKeyWithPath, order) {
        if (objectKeyWithPath === void 0) { objectKeyWithPath = ''; }
        if (order === void 0) { order = 'asc'; }
        var predicatePathValues = objectKeyWithPath.split('.');
        var desc = order === 'desc';
        if (!objectKeyWithPath) {
            return desc ? values.sort().reverse() : values.sort();
        }
        return values.sort(function (a, b) {
            var result = 0;
            var leftSideValue = a;
            var rightSideValue = b;
            predicatePathValues.forEach(function (predicate) {
                leftSideValue = leftSideValue[predicate];
                rightSideValue = rightSideValue[predicate];
            });
            if (leftSideValue < rightSideValue) {
                result = desc ? 1 : -1;
            }
            else if (rightSideValue < leftSideValue) {
                result = desc ? -1 : 1;
            }
            return result;
        });
    };
    OrderByPipe = __decorate([
        Pipe({ name: 'orderBy' })
    ], OrderByPipe);
    return OrderByPipe;
}());

var OrderByModule = /** @class */ (function () {
    function OrderByModule() {
    }
    OrderByModule = __decorate([
        NgModule({
            declarations: [OrderByPipe],
            exports: [OrderByPipe],
            providers: [OrderByPipe]
        })
    ], OrderByModule);
    return OrderByModule;
}());

/**
 * Generated bundle index. Do not edit.
 */

export { OrderByModule, OrderByPipe };
