// ngx-order-by/src/app/ngx-order-by/ngx-order-by.pipe.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { Pipe } from '@angular/core';
var NgxOrderByPipe = (function () {
    function NgxOrderByPipe() {
    }
    NgxOrderByPipe.prototype.transform = function (values, fields, reverse, changeIndicator) {
        this.fields = fields.slice();
        this.reverse = reverse;
        values.sort(this.compare.bind(this));
        if (reverse) {
            values.reverse();
        }
        return values;
    };
    NgxOrderByPipe.prototype.compare = function (a, b) {
        var result = 0, i = 0;
        while (result === 0 && i < this.fields.length) {
            result = this.dynamicSort(a, b, this.fields[i]);
            i++;
        }
        return result;
    };
    NgxOrderByPipe.prototype.dynamicSort = function (a, b, property) {
        return a[property] > b[property] ? 1 : a[property] < b[property] ? -1 : 0;
    };
    return NgxOrderByPipe;
}());
NgxOrderByPipe = __decorate([
    Pipe({
        name: 'ngxOrderBy',
    })
], NgxOrderByPipe);
export { NgxOrderByPipe };
